### PR TITLE
First version of the IRCv2 implementation

### PR DESCRIPTION
**Pull request by ztamas83**
I have started the implementation of the IntelligentRoomController v2 against the Home-Assistant climate entity.
With this version the changes are synced both ways between HA and Loxone.

This is my very first work in HA so all comments and corrections are welcome.

### CODE REVIEW

File: custom_components/loxone/climate.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+"""
+import logging
+import collections
+from typing import Any
+from voluptuous import All, Range, Optional
+
+from homeassistant.components.climate.const import (
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_AWAY,
```
Comment (JoDehli) [inline custom_components/loxone/climate.py:12]: These const are not needed:
   
 PRESET_COMFORT,
    PRESET_ECO,
    PRESET_AWAY,

File: custom_components/loxone/climate.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
+"""
+Loxone cover component.
+"""
+import logging
+import collections
+from typing import Any
```
Comment (JoDehli) [inline custom_components/loxone/climate.py:6]: These imports are not needed:
import collections
from typing import Any

File: custom_components/loxone/climate.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    ATTR_HVAC_MODE,
+    ClimateEntity,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    PLATFORM_SCHEMA
+)
+
+from homeassistant.const import (
+    CONF_VALUE_TEMPLATE, CONF_NAME)
+from homeassistant.helpers.event import track_utc_time_change
```
Comment (JoDehli) [inline custom_components/loxone/climate.py:33]: track_utc_time_change import is not needed

File: custom_components/loxone/climate.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+#SUPPORT_SWING_MODE = 64
+#SUPPORT_TARGET_HUMIDITY = 128
+#SUPPORT_TARGET_TEMPERATURE = 256
+#SUPPORT_TARGET_TEMPERATURE_RANGE = 512
+#other support values 8, 16...
+
+async def async_setup_platform(hass, config, async_add_devices, discovery_info={}):
+    value_template = config.get(CONF_VALUE_TEMPLATE)
+    name = config.get(CONF_NAME)
+    autoMode = 0 if config.get(CONF_HVAC_AUTO_MODE) is None else config.get(CONF_HVAC_AUTO_MODE)
```
Comment (JoDehli) [inline custom_components/loxone/climate.py:79]: autoMode variable should be lowercase for example: auto_mode

File: custom_components/loxone/climate.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        """Return a list of available preset modes.
+
+        Requires SUPPORT_PRESET_MODE.
+        """
+        return [mode['name'] for mode in self._modeList]
+
+    def set_hvac_mode(self, hvac_mode: str):
+        """Set new target hvac mode."""
+
+        targetMode = self.autoMode if hvac_mode == HVAC_MODE_AUTO else OPMODETOLOXONE[hvac_mode]
```
Comment (JoDehli) [inline custom_components/loxone/climate.py:229]: targetMode should be lowercase for example target_mode

**Review by JoDehli — CHANGES_REQUESTED** [review]
Only a view format and variables names should be changed. Again great work. Thanks.

What I must do is also use the new f-string notation. Looks much better :-)

File: custom_components/loxone/climate.py (hunk 2 above)
Comment (JoDehli) [reply custom_components/loxone/climate.py:6]: of course logging must stay!

**Comment by ztamas83** [comment]
Two config things I don't really know how to solve:
1) Different controllers in different rooms may have different capabilities (HVAC modes), users perhaps should be able to set the capabilities in config instead of automatically add all units with all capabilities? E.g. I only have heater control, should not be able to set HVAC_COOL
2) Same for the auto mode, if i want Auto heat/cool, or just auto heat or auto cool. For this I already added the config but I'm not sure how to add the config into HASS. Need to figure it out.

Maybe they are for the next release if the basic IRCv2 functionality is so awaited :)

I'll push the changes you commented on.

**Comment by JoDehli** [comment]
Thanks again for your work. I am glad that somebody helps me. Because I have not so much time and a view other projects I am working on. I merged it and made some smaller changes. It is only because my IDE complaints about some PEP errors. For example some variables names. And I changed the name of the RoomController to RoomControllerV2. 

I must think about your questions. For now I have no solutions. But I think we have time to improve it. I wanted to release it because some users are already requested the RoomController. But I think the best would be to get the information automatically from the loxone. I do not know it but does the official Loxone app recognize if a entity only have a heating or not? The same information should also be available for us.

**Comment by MaartenMees** [comment]
Thanks Guys!!!

**Pull request merged**